.github/workflows/main.yml

```diff
@@ -73,13 +73,59 @@ jobs:
           python3.8 -m pip install 'virtualenv<20.22' 'tox==4.5.1' tox-gh-actions
           tox --workdir .github/workflows/.tox --recreate
 
-      - name: Upload coverage reports to Codecov
-        if: ${{ matrix.os == 'ubuntu-20.04' && github.actor != 'nektos/act'}}
+
+      # The new reliable Codecov upload requires Codecov to query the GitHub API to check
+      # the repo and the commit. The repo (or organisation) owner needs to login to
+      # codev, generated the CODECOV_TOKEN and save it as a secret in the ORG or the repo:
+      # https://docs.codecov.com/docs/adding-the-codecov-token
```

Comment:
- The old tokenless Codecov upload is unreliable: GitHub rate-limits unauthenticated API calls.
- The only solution is to login to codecov.io, generate a token, save it as repo/org secret.
- This needs to be done by the repo owner or .org owner, who might be reluctant, because of the scary warnings shown by GitHub about allowing `codecov.io` to `impersonate` (authenticate only our behalf), but it's needed to make the upload reliable.
- This part of the PR prepares that CI picks up the token and uses it once it is set.